doc/api/configuration.md

```diff
@@ -241,3 +241,33 @@ print(res)
 res = %sql SELECT * FROM languages LIMIT 2
 print(res)
 ```
+
+## `persist_snippets`
+
+```{versionadded} 0.8.1
+```
+
+Default: `False`
+
+Stores the snippet in a SQL file.
+
+```{code-cell} ipython3
+%config SqlMagic.persist_snippets = True
+```
+
+```{code-cell} ipython3
+%%sql --save rated_high
+select * from languages where rating > 10.5
+```
+
+The sql file will be saved at: 
+
+`jupysql-snippets/rated_high.sql` 
+
+with the following content: 
+
+`select * from languages where rating > 10.5`
+
+
+JupySqL will automatically load the snippets from the SQL files when re-initializing the magic.
+It will remain readily available for future use.
```

Comment:
is this true? If I modify the SQL files, I don't think the next time I open my notebook the cells will be updated, right? you only implemented one-way-sync (which is fine): from the notebook to the SQL file

Reply:
So when we open a new notebook, and are initialising the magic, we check if the jupysql-snippets directory exists. If it does we load the contents of the sql files in it as snippets. 

If we change the .sql file manually, and then reload the notebook, the edited sql version is loaded as the new snippet.